backend/generator.js

```diff
@@ -3,6 +3,10 @@ var exports = module.exports = {};
 var uuidV4 = require("uuid/v4");
 
 exports.executeScript = function (socket, formData) {
+  if (!isValid(formData)) {
```

Comment:
I can see that but still the function almost duplicated. That's not a good idea. Maybe return a error code from the function and use it to display message in the front end. That way you'll only need to define the function once.